providers/dns/linodev4/linodev4.go

```diff
@@ -97,19 +99,22 @@ func NewDNSProviderConfig(config *Config) (*DNSProvider, error) {
 
 // Timeout returns the timeout and interval to use when checking for DNS
 // propagation.  Adjusting here to cope with spikes in propagation times.
-func (d *DNSProvider) Timeout() (timeout, interval time.Duration) {
-	// Since Linode only updates their zone files every X minutes, we need
-	// to figure out how many minutes we have to wait until we hit the next
-	// interval of X.  We then wait another couple of minutes, just to be
-	// safe.  Hopefully at some point during all of this, the record will
-	// have propagated throughout Linode's network.
-	minsRemaining := dnsUpdateFreqMins - (time.Now().Minute() % dnsUpdateFreqMins)
-
-	timeout = (time.Duration(minsRemaining) * time.Minute) +
-		(minTTL * time.Second) +
-		(dnsUpdateFudgeSecs * time.Second)
-	interval = d.config.PollingInterval
-	return
+func (d *DNSProvider) Timeout() (time.Duration, time.Duration) {
+	timeout := d.config.PropagationTimeout
+	if d.config.PropagationTimeout <= 0 {
```

Comment:
Why not this instead?

```suggestion
	if timeout <= 0 {
```